asl_xdsl/dialects/asl.py

```diff
@@ -827,6 +827,217 @@ def __init__(
         )
 
 
+@irdl_op_definition
+class ZeroExtendBitsOp(IRDLOperation):
+    """A bit vector zero-extend operation."""
+
+    name = "asl.zero_extend_bits"
+
+    S: ClassVar = VarConstraint("S", BaseAttr(BitVectorType))
+    T: ClassVar = VarConstraint("T", BaseAttr(BitVectorType))
+
+    lhs = operand_def(S)
+    rhs = operand_def(IntegerType())
+    res = result_def(T)
```

Comment:
You can actually just replace this with
```suggestion
    lhs = operand_def(BitVectorType)
    rhs = operand_def(IntegerType())
    res = result_def(BitVectorType)
```
The VarConstraint are only really used whenever you need two types to be exactly equal